lib/Conversion/HWArithToHW/HWArithToHW.cpp

```diff
@@ -302,9 +302,10 @@ struct ICmpOpLowering : public OpConversionPattern<ICmpOp> {
     Value rhsValue = extendTypeWidth(rewriter, loc, adaptor.getRhs(), cmpWidth,
                                      rhsType.isSigned());
 
-    auto newOp = rewriter.replaceOpWithNewOp<comb::ICmpOp>(
-        op, combPred, lhsValue, rhsValue, false);
+    auto newOp = rewriter.create<comb::ICmpOp>(op->getLoc(), combPred, lhsValue,
+                                               rhsValue, false);
     newOp->setDialectAttrs(op->getDialectAttrs());
```

Comment:
Do we have to wrap this with `rewriter.modifyOpInPlace`?

Reply:
Yes, it would be necessary.

Reply:
Done in https://github.com/llvm/circt/pull/6960/commits/097d0c75e586abeb15252e3ca120fdd71f2e1429